querybook/webapp/components/ExecutionRangeField/ExecutionRangeField.tsx

```diff
@@ -0,0 +1,291 @@
+import React, { useReducer, useCallback } from 'react';
+import moment from 'moment';
+import styled from 'styled-components';
+
+import type { IOccurence, IDate } from 'const/schedule';
+
+import { SearchDatePicker } from 'components/Search/SearchDatePicker';
+import { FormField } from 'ui/Form/FormField';
+import { TimePicker } from 'ui/TimePicker/TimePicker';
+import { Tabs } from 'ui/Tabs/Tabs';
+import NumberInput from 'ui/NumberInput/NumberInput';
+import { StyledButton } from 'ui/Button/StyledButton';
+
+import { reducer } from './reducer';
+
+const TimerDiv = styled.div`
+    text-decoration: underline;
+    cursor: pointer;
+`;
+
+const DatePickerWrapper = styled.div`
+    display: flex;
+`;
+
+const TimePickerWrapper = styled(TimePicker)`
+    margin-left: 10px;
+`;
+
+const TabsWrapper = styled.div`
+    padding-bottom: 10px;
+`;
+
+const NumberInputWrapper = styled(NumberInput)`
+    max-width: 41%;
+    margin-bottom: 10px;
+    display: block;
+`;
+
+const StyledBtn = styled(StyledButton)`
+    background-color: var(--bg-light);
+`;
+
+function formatDateTime(date, prevValue) {
+    return moment(date)
+        .hours(prevValue.hours())
+        .minutes(prevValue.minutes())
+        .format('X');
+}
+
+function formatTimePickerValue(value) {
+    return value
+        ? moment(value, 'X').local()
+        : moment(new Date(), 'X').minutes(0).hours(0);
+}
+
+function formatTimeLabel(value) {
+    return value
+        ? `${moment(value, 'X').local().format('DD.MM.YYYY HH:mm')}`
+        : '--:--';
+}
+
+function generateRangeLabel(timeRange) {
+    if (timeRange.occurences) {
+        return `${timeRange.occurences} time(s)`;
+    }
+
+    return `${formatTimeLabel(timeRange.startTime)} -
+${formatTimeLabel(timeRange.endTime)}`;
+}
+
+function formatDefaultDateValue(value) {
+    const defaultValue = value ?? moment().format('X');
+    return moment(defaultValue, 'X');
+}
+
+const tabs = [
+    { name: 'Time range', key: 'timeRange' },
+    { name: 'occurences', key: 'occurences' },
+];
+
+interface IExecutionRangeFieldProps {
+    updateValues: (values: IDate | IOccurence) => void;
+    values: IDate | IOccurence;
+}
+
+export const ExecutionRangeField: React.FunctionComponent<IExecutionRangeFieldProps> = ({
```

Comment:
can you rename this to ScheduleRangeField since its not related to execution here?